src/views/pages/Resources.vue

```diff
@@ -264,10 +268,15 @@ const createResource = async (): Promise<void> => {
     category: openTab.value,
     description: "",
   } as Resource).then(() => {
-    newResource.value.name = "";
-    newResource.value.creating = false;
-
-    updateResources();
+    updateResources().then(() => {
+      resources.value.forEach((resource) => {
+        if (resource.name === newResource.value.name) {
```

Comment:
This might not be reliable as names are not guaranteed to be unique. Would need to evaluate the createOrUpdateResource return and use the `id` field.

Reply:
The names have to be unique as the document gets renamed to be the name of the file. So if anything the API needs to not allow the creation of documents with the same name.

Reply:
You're right.. I've adjusted the API to add the ID to the slug to ensure uniqueness https://github.com/adh-partnership/api/pull/111 -- this is a non-breaking change unless people directly link to the CDN as any future upload will now have the ID in it.

Reply:
in your .then you can utilize .then((resource) as the return IS the created or updated resource. https://github.com/adh-partnership/frontend/blob/main/src/utils/api.ts#L32